### PR TITLE
Adding batch processing, cleanup, and content length.

### PR DESCRIPTION
**Pull request by stratease**
Adding a batch processing functionality to the frontend. Will process 50 immediately, and queues the remainder with WP cron recursively.
Also added a content length field for event post content (to generate large content strings for memory/perf testing).
Also did some API cleanup - hopefully for the better!

I want to test it a bit more, ran out of time - but so far it seems solid, would love eyes on it.

### CODE REVIEW

File: src/Tribe/Generator/Event.php [hunk 1]
```diff
@@ -51,14 +51,15 @@ public function create( $quantity = 1, array $args = [], callable $tick = null )
 		$event_cat_arg  = isset( $args['event_category'] ) ? Arr::list_to_array( $args['event_category'] ) : [];
 		$custom_tag_arg = isset( $args['custom_tag'] ) ? array($args['custom_tag']) : [];
 		$event_tag_arg  = isset( $args['event_tag'] ) ? Arr::list_to_array( $args['event_tag'] ) : [];
+		$content_length =$args['content_length'] ??  null;
```
Comment (lucatume) [inline src/Tribe/Generator/Event.php]:
```suggestion
		$content_length = $args['content_length'] ??  null;
```

File: src/Tribe/Generator/Event.php [hunk 2]
```diff
@@ -120,21 +121,26 @@ public function create( $quantity = 1, array $args = [], callable $tick = null )
...
+	 * @since TBD Adding content length param.
+	 *
 	 * @return string[]
 	 */
 	public function random_event_data(
 		$from_date, $to_date, $is_featured, $is_virtual,
 		$is_recurring, $recurring_type,
-		$event_category, $event_tag
+		$event_category, $event_tag,
+		$content_length
```
Comment (lucatume) [inline src/Tribe/Generator/Event.php:137]: The `$content_length` parameter might be `null`: the method should allow for it.

**Review by lucatume — CHANGES_REQUESTED** [review]
So far, there are some smaller changes to do.

I _guess_ the `tec_ext_test_data_generator_handle_batch` would be the one fired in Cron context, but cannot see where it's hooking on the crown.

Question: why Cron? This plugin assumes TEC and TEC contains Common that contains Action Scheduler, why not use that?

**Comment by stratease** [comment]
@lucatume 

> I _guess_ the `tec_ext_test_data_generator_handle_batch` would be the one fired in Cron context, but cannot see where it's hooking on the crown.

That is here ``` wp_schedule_single_event( time() + 66, 'tec_ext_test_data_generator_handle_batch' ... ```

It will recurse on itself until all "queued" `quantity` is decremented.

> Question: why Cron? This plugin assumes TEC and TEC contains Common that contains Action Scheduler, why not use that?

I want fewer dependencies. Cron is core, this adds one more connection and complexity to our infrastructure in TEC. Don't think it's necessary?

File: src/Tribe/Generator/Event.php (hunk 2 above)
Comment (stratease) [reply src/Tribe/Generator/Event.php:137]: Not sure what you mean by that - it does allow for it...?

Comment (lucatume) [reply src/Tribe/Generator/Event.php:137]: Sorry, I was not very precise.
[Here](https://github.com/mt-support/tribe-ext-test-data-generator/pull/31/files#diff-2755a9c17280041a2cc7e87f0ed0a63e671863372124cbe5ef172cf929084da1R54) `$content_length` is read from the input arguments, or set to `null`.
[Here](https://github.com/mt-support/tribe-ext-test-data-generator/pull/31/files#diff-2755a9c17280041a2cc7e87f0ed0a63e671863372124cbe5ef172cf929084da1R62) that possibly `null` value is passed to the `random_event_data` method.
The `random_event_data` method is not accepting a `null` value, though.

**Comment by lucatume** [comment]
> I want fewer dependencies. Cron is core, this adds one more connection and complexity to our infrastructure in TEC. Don't think it's necessary?

Being Action Scheduler part of Common and TEC, I would take that for granted and it's more reliable than Cron when it comes to background execution.
Action Scheduler and Cron work the same way dispatching an action at some point, with different systems, why not check for the `as_*` functions existence and use Action Scheduler, if available?

**Comment by stratease** [comment]
Added AS. Going to merge.

**Pull request merged**